### ISSUE TITLE
dotnetcore support

### ISSUE THREAD

**Issue by flytzen** [uglybugger/NotDeadYet#7]
Any thoughts on whether it'd be worth adding dotnetcore support? 
It's sort-of separate from adding aspnetcore support.

**Comment by uglybugger** [comment]
Has dotnetcore stopped being a moving target yet? ;)

I'm happy for anyone to have a go at it but I'm not personally going to spend any meaningful effort on anything dotnetcore-related until NUnit, ReSharper and TeamCity all support it nicely.

**Comment by flytzen** [comment]
I think there are a few parts to it. 
1) Change it to use project.json and .xproj. This may change how the Nuget packages are built so may impact the publish pipeline, depending on how the publishing to Nuget is currently handled?
2) Change the core assembly to "netstandard" as well as the full framework. The other projects should be able to remain as they are.
3) Maybe, possibly, do a plugin for ASP.Net Core, similar to the MVC one, but not sure it's worth it.

I'd be happy to do the code changes for 1 and 2 - it should be pretty straight forward and I've done a fair bit of work in that space recently. But I'm not sure about the impact on the build/publish pipeline so would need a bit of advice on that. How do you publish to to Nuget at the moment?

**Comment by flytzen** [comment]
I've had a bit more of a look into this. I can see a way to do it - but the forthcoming move away from xproj and back to csproj may make it worthwhile to leave it for the moment.

**Comment by grokky1** [comment]
@flytzen That has happened now. Did you find a way to make it work with Core?

**Comment by flytzen** [comment]
I didn't, but I was just thinking the other day that I should give it a go, so happy to do that.
Would you be okay with an update to VS2017?

**Comment by grokky1** [comment]
@flytzen I still haven't made the move to VS2017... Guess it's time. 😆

**Comment by flytzen** [comment]
Just FYI; work has started. Looking quite straight forward so far on the *general* porting. 
However, it relies in turn on ThirdDrawer which will also have to be updated in order to provide proper core support (or dependency remove if it's only minor, I'll look).

Only real question is about how you currently *pack*? I assume a manual Nuget Pack with the version numbers on the command line? If you use an automatic service, please let me know which.
It'll be slightly different with "dotnet pack", but only a little. I'll include some sample build files you can use and adjust to your workflow.

**Comment by flytzen** [comment]
Update (@grokky1)  : Everything is now running on .Net core and there is an ASP.Net Core sample app. Current work is here: https://github.com/flytzen/NotDeadYet/tree/DotNetCore.
I had to remove the dependency on ThirdDrawer.

Main outstanding task is to remove the nuspec files and set up a simple package generation script.

There is one small item that stops it being completely done; NUnit doesn't currently work with Core and VS2017. The simplest short-term fix would be to convert the unit tests to XUnit, *but not sure how you feel about that, @uglybugger?* That should allow the tests to run with both the frameworks as well.

Of course, MS are rather pissing on this parade with this little gem: https://github.com/aspnet/HealthChecks

**Comment by grokky1** [comment]
Haven't tried in VS2017. But NUnit definitely works with .NET Core.

**Comment by flytzen** [comment]
@grokky1 Yes, I'm talking about this issue with the runner and the new csproj format: https://github.com/nunit/dotnet-test-nunit/issues/108. Sorry for not being clear.

**Comment by grokky1** [comment]
Wow what a mess... It's not their fault though. Glad I'm still on vs2015 then 😆

**Comment by ChrisMcKee** [comment]
nunit works fine in VS2017 you just have to install an extra package; regardless though you can just leave the test project pointing at .net 4.6, it works fine.

**Comment by graemechristie** [comment]
Is there any activity on this ? .NEt core is now 2.0 with 2.1 around the corner. Is this project likely to be maintained going forward ?

**Comment by flytzen** [comment]
@graemechristie It appears to be dead, sadly.

**Comment by Jaxelr** [comment]
@flytzen not anymore, check Issue #17

**Comment by flytzen** [comment]
@Jaxelr @uglybugger Ah, cool stuff :)
Do you want to review the PR for the migration to .Net Core? It's currently targeting netstandard16 but could easily be changed to netstandard20 if you prefer. Only outstanding thing is NUnit and the build process, the latter is something we'd need to talk about as it depends on how you want to run that.
